packages/nestjs/src/setup.ts

```diff
@@ -0,0 +1,156 @@
+import type {
+  ArgumentsHost,
+  CallHandler,
+  DynamicModule,
+  ExecutionContext,
+  NestInterceptor,
+  OnModuleInit,
+} from '@nestjs/common';
+import { Catch } from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
+import { Global, Module } from '@nestjs/common';
+import { APP_FILTER, BaseExceptionFilter } from '@nestjs/core';
+import { APP_INTERCEPTOR } from '@nestjs/core';
+import {
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  captureException,
+  getClient,
+  getDefaultIsolationScope,
+  getIsolationScope,
+  spanToJSON,
+} from '@sentry/core';
+import type { Span } from '@sentry/types';
+import { logger } from '@sentry/utils';
+import type { Observable } from 'rxjs';
+
+/**
+ * Note: We cannot use @ syntax to add the decorators, so we add them directly below the classes as function wrappers.
+ */
+
+/**
+ * Interceptor to add Sentry tracing capabilities to Nest.js applications.
+ */
+export class SentryTracingInterceptor implements NestInterceptor {
+  /**
```

Comment:
h: I have a stupid hunch about how we order use of the decorators and exporting. Can we do it like the following so we ensure that we are acutally exporting the decorated versions of the classes:


```ts
class SentryTracingInterceptor implements NestInterceptor {
    // ...
}
Injectable()(SentryTracingInterceptor);
export { SentryTracingInterceptor }
```

I am just slightly worried that at one point the class is turned immutable (or is frozen) when exported, and decorating won't do anything. I think our tests prove that this is not necessary right now but just logically and defensively it is more sound.

Reply:
Sure, updated